### PR TITLE
Add TLS for all devices

### PR DESCRIPTION
**Pull request by goligo**
Changes in genpkglist.py/site.mk have caused many devices to build without TLS. As we dropped 4/32 devices completely in this release, all devices should have sufficient flash to include TLS.

This change includes TLS for all firmware images, except for TP-Link Range Extender devices, which do not have sufficient flash space availavble.

### CODE REVIEW

**Comment by GoliathLabs** [comment]
I'd let the CI run for this change

**Review by grische — COMMENTED** [review]
Why are we changing the logic of genpkglist.py? We are pretty much in sync with upstream right now, do we want to upstream our changes again later?

File: contrib/genpkglist.py [hunk 1]
```diff
@@ -71,36 +71,24 @@ def exclude(self, devices: [str], pkglists: [PackageList]=None):
         return self
 
     def render(self):
-        if bool(self.includes):
```
Comment (GoliathLabs) [inline contrib/genpkglist.py:74]: Why has this been removed? Without this we can include and exclude a package list at the same time which shouldn't be possible

**Review by GoliathLabs — DISMISSED** [review]

**Comment by goligo** [comment]
Logic of genpkglist.py was changed, because it was insufficient, there was no easy way to exclude TLS only for the specific devices where it did not fit. As it already has been merged this way in stable, I would keep it for now and look into this again with the next update of genpkglist.

File: contrib/genpkglist.py (hunk 1 above)
Comment (goligo) [reply contrib/genpkglist.py:74]: Yes, it was removed intentionally to allow to include and exclude at the same time.

**Comment by GoliathLabs** [comment]
> Logic of genpkglist.py was changed, because it was insufficient, there was no easy way to exclude TLS only for the specific devices where it did not fit. As it already has been merged this way in stable, I would keep it for now and look into this again with the next update of genpkglist.

Is there a better approach for this so we don't have to worry about the inconsistencies caused by including and excluding a package list at the same time?

File: contrib/genpkglist.py (hunk 1 above)
Comment (goligo) [reply contrib/genpkglist.py:74]: The alternative would have been, for the targets where the new include option was used, to switch to exclude again and list ALL the other (not included) devices in the genpkglist file.

**Comment by goligo** [comment]
> > Logic of genpkglist.py was changed, because it was insufficient, there was no easy way to exclude TLS only for the specific devices where it did not fit. As it already has been merged this way in stable, I would keep it for now and look into this again with the next update of genpkglist.
> 
> Is there a better approach for this so we don't have to worry about the inconsistencies caused by including and excluding a package list at the same time?

Which inconsistencies are you talking about? I guess whoever decided to make these two options exclusive couldn't think of a use case to have both. Well, we have such a use case.

File: contrib/genpkglist.py (hunk 1 above)
Comment (GoliathLabs) [reply contrib/genpkglist.py:74]: Sounds fair

**Review by GoliathLabs — COMMENTED** [review]
Sound fair enough for me now (as we have to deal with those TP-Link RE devices)

**Comment by goligo** [comment]
Obwohl ich TLS rausgenommen habe, hängt er immer noch beim gleichen Device...
```
2022-10-27T02:08:33.3913174Z [ -f /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/dlink_dir-825-b1-kernel.bin -a -f /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/root.squashfs+pkg=257eab6e ]
2022-10-27T02:08:33.3928648Z dd if=/home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/dlink_dir-825-b1-kernel.bin >> /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin
2022-10-27T02:08:33.4071103Z 4495+1 records in
2022-10-27T02:08:33.4071684Z 4495+1 records out
2022-10-27T02:08:33.4072256Z 2301703 bytes (2.3 MB, 2.2 MiB) copied, 0.0120887 s, 190 MB/s
2022-10-27T02:08:33.4074085Z dd if=/home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/root.squashfs+pkg=257eab6e >> /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin
2022-10-27T02:08:33.4387444Z 7945+1 records in
2022-10-27T02:08:33.4387661Z 7945+1 records out
2022-10-27T02:08:33.4387977Z 4067958 bytes (4.1 MB, 3.9 MiB) copied, 0.0291664 s, 139 MB/s
2022-10-27T02:08:33.4395460Z /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/staging_dir/host/bin/padjffs2 /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin  64
2022-10-27T02:08:33.4418018Z padding image to 00620000
2022-10-27T02:08:33.4455168Z bash: line 0: printf: write error: Transport endpoint is not connected
2022-10-27T02:08:33.4476016Z echo '{  "metadata_version": "1.1", "compat_version": "1.0",   "supported_devices":["dlink,dir-825-b1","dir-825-b1"], "version": { "dist": "OpenWrt", "version": "SNAPSHOT", "revision": "r20729+6-579703f38c", "target": "ath79/generic", "board": "dlink_dir-825-b1" } }' | fwtool -I - /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin
2022-10-27T02:08:33.4501788Z Failed to open firmware file
2022-10-27T02:08:33.4505837Z bash: line 0: echo: write error: Broken pipe
2022-10-27T02:08:33.4510286Z sha256sum "/home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin" | cut -d" " -f1 > "/home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin.sha256sum"
2022-10-27T02:08:33.4511698Z make[8]: [Makefile:105: /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin] Error 1 (ignored)
2022-10-27T02:08:33.4546627Z sha256sum: /home/runner/work/site-ffm/site-ffm/gluon-build/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/tmp/openwrt-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin: No such file or directory
```

**Comment by goligo** [comment]
Die Größe der Dateien ist auch identisch zum vorherigen Run, als ob er die Änderung gar nicht angenommen hätte?

**Comment by goligo** [comment]
Lokaler Build läuft, das heisst CI-Build ist kaputt. Ich pushe nochmal irgendwas in der Hoffnung, dass er sich dann wieder einkriegt.

**Review by GoliathLabs — APPROVED** [review]

**Pull request merged**